### ISSUE TITLE
Harmonize python repos to 3.11

### ISSUE THREAD

**Issue by rgaudin** [openzim/overview#13]
For the sake of harmonization and ease of dev/maint, we want to use a single major Python version as base for all our python projects.

Following https://en.wikipedia.org/wiki/History_of_Python#Table_of_versions, we want to use `3.11` as of now.
As such, support for python `3.6` (not receiving secu updates) and `3.7` (secu updates to end in 6m) will be dropped from those projects.

**Comment by rgaudin** [comment]
Kind of the point of this ticket to figure this out.

List of repos:
- https://github.com/openzim/overview/wiki/Scrapers-list including tools
- https://github.com/offspot/?q=&type=all&language=python&sort=

What's on kiwix doesn't really fall into the python-repo category.

What I have in mind:
- Make sure it works. 3.6 code is mostly forward compatible except a couple changed API like with `typing`. This would mostly mean creating a new 3.11 environment and make sure that dependencies are compatible and can be installed ; potentially updating some.
- update workflows to run actions on 3.8-3.11
- Update docker image to use a an image with the appropriate python version
- Update descriptions/readme/changelog to reflect the change

**Comment by kelson42** [comment]
I propose as well to use that badge ![PyPI - Python Version](https://img.shields.io/pypi/pyversions/libzim)

**Comment by benoit74** [comment]
I realize that I've probably misunderstood this PR so far; I've understood that we want to support only one single Python version (3.11 here) while it looks like the original idea was more to support a range of active versions (3.8 to 3.11 originally) and deploy (Docker) with on single version (3.11).

Should I revert some changes so that scrapers codes are still compatible with 3.8 and 3.9? 

Now that Docker is mostly mainstream, I don't see a huge benefit in supporting 3 Python versions for scrapers which are mostly always "end-products", while it obviously limit what is possible with code, and since we usually have no automated tests for scrapers I'm pretty sure there will be some errors which won't be catch.

But maybe I'm too pessimistic + a bit lazy ^^

Nota: now that 3.12 is out, I'm aware we will very soon switch to 3.9 to 3.12 once scraperlib is ready

**Comment by rgaudin** [comment]
@benoit74 [RTFM](https://github.com/openzim/_python-bootstrap/wiki/pyproject.toml#target-python-version) 😅

**Comment by benoit74** [comment]
Thank you for your very good memory! As usual I forgot about the manual, there is just so many 🤣 Good news is that I was right in my changes (I knew it came from somewhere), latest single version is ok for scrapers, it was just not reflected in this issue (and I became worried for nothing, but I'm better safe than sorry that been said 😉)

**Comment by rgaudin** [comment]
Yes I think this issue predates the bootstrap and our knowledge management is organic 😉

**Comment by kelson42** [comment]
This issue might desserve an update

**Comment by rgaudin** [comment]
Closing as this ticket brings more confusion than solution. Our strategy is now quite clearly defined and understood. There are some projects that don't receive enough maintenance and are still using unsupported versions. Individual ticket would better suit the need.
